### ISSUE TITLE
Sequence for primary key column results in TransactionContext Error

### ISSUE THREAD

**Issue by Tmonster** [duckdb/duckdb#6421]
### What happens?

When copying a table to include a new primary key based on a sequence, the following error is produced
```
Error: TransactionContext Error: Failed to commit: Constraint Error: PRIMARY KEY or UNIQUE constraint violated: duplicated key
```

### To Reproduce

```
call dbgen(sf=5);
CREATE TABLE lineitem_1_with_pk
(
    l_orderkey    BIGINT not null,
    l_partkey     BIGINT not null,
    l_suppkey     BIGINT not null,
    l_linenumber  BIGINT not null,
    l_quantity    DOUBLE PRECISION not null,
    l_extendedprice  DOUBLE PRECISION not null,
    l_discount    DOUBLE PRECISION not null,
    l_tax         DOUBLE PRECISION not null,
    l_returnflag  CHAR(1) not null,
    l_linestatus  CHAR(1) not null,
    l_shipdate    DATE not null,
    l_commitdate  DATE not null,
    l_receiptdate DATE not null,
    l_shipinstruct CHAR(25) not null,
    l_shipmode     CHAR(10) not null,
    l_comment      VARCHAR(44) not null,
    l_tmp_primary_key BIGINT PRIMARY KEY
);
CREATE SEQUENCE l1_pk START 1;
INSERT INTO lineitem_1_with_pk SELECT l1.*, nextval('l1_pk') FROM lineitem l1; -- this line produces the error
```
If I remove the primary key constraint and insert the data, no error is generated, and all l_tmp_primary_key values are unique.

### OS:

MacOS M1

### DuckDB Version:

v0.7.1-dev229

### DuckDB Client:

CLI

### Full Name:

Tom Ebergen

### Affiliation:

DuckDB Labs

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the report! I can't seem to reproduce this - is this a consistent issue, or does it pop up sporadically? Does it require a persistent database?

**Comment by Tmonster** [comment]
hmm, very weird. The issue is consistent, and you shouldn't need a persistent database. I just replicated the issue again after merging with master and feature. 
Was also able to reproduce in release and debug. I'll try one more time with a fresh feature branch

**Comment by Tmonster** [comment]
Also just saw this error message when exiting the CLI in debug mode 
```
libc++abi: terminating with uncaught exception of type duckdb::InternalException: INTERNAL Error: Assertion triggered in file "/Users/tomebergen/duckdb/src/storage/buffer_manager.cpp" on line 26: size == 0
[1]    33947 abort      ../duckdb/build/debug/duckdb
```
Don't know how much it helps though

**Comment by Mytherin** [comment]
Might have something to do with your memory limit, what is it set to?

**Comment by Tmonster** [comment]
Ah you are right. You have to set your memory limit to 16GB to reproduce this error

**Comment by Mytherin** [comment]
This has been fixed in #6693
